modules/collect/src/main/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeries.java

```diff
@@ -0,0 +1,984 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.collect.timeseries;
+
+import static com.opengamma.collect.Guavate.toImmutableList;
+import static com.opengamma.collect.timeseries.DenseLocalDateDoubleTimeSeries.DenseTimeSeriesCalculation.SKIP_WEEKENDS;
+import static java.time.temporal.ChronoUnit.DAYS;
+
+import java.io.Serializable;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.ChronoField;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.OptionalDouble;
+import java.util.Set;
+import java.util.function.DoubleBinaryOperator;
+import java.util.function.DoubleUnaryOperator;
+import java.util.function.ObjDoubleConsumer;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedMap;
+import com.opengamma.collect.ArgChecker;
+import com.opengamma.collect.function.ObjDoublePredicate;
+import com.opengamma.collect.tuple.Pair;
+
+/**
+ * An immutable implementation of {@code DoubleTimeSeries} where the
+ * data stored is expected to be dense. For example, points for every
+ * working day in a month. If sparser data is being used then
+ * {@link SparseLocalDateDoubleTimeSeries} is likely to be a better
+ * choice for the data.
+ * <p>
+ * This implementation uses arrays internally.
+ */
+@BeanDefinition(builderScope = "private")
+public class DenseLocalDateDoubleTimeSeries
```

Comment:
General comment, class is hard to review as there is nothing to break up the stream of methods (horizontal lines). Also, the private helper methods could do with being more organized, notably validIndices() and isValidPoint().